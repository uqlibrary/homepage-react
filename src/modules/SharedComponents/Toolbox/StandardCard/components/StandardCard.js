import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';
import { HelpIcon } from '../../HelpDrawer';
import { withStyles } from '@material-ui/core/styles';

export const styles = theme => ({
    card: {
        overflow: 'unset',
        fontWeight: theme.typography.fontWeightRegular,
    },
    cardContentNoPadding: {
        paddingTop: '0px !important',
        paddingBottom: '0px !important',
        paddingLeft: '0px !important',
        paddingRight: '0px !important',
    },
    cardHeaderPrimary: {
        color: theme.palette.white.main,
        overflow: 'hidden',
        whiteSpace: 'nowrap',
        textOverflow: 'ellipsis',
        backgroundColor: theme.palette.primary.main,
        borderRadius: '4px 4px 0px 0px',
        padding: '12px 24px',
    },
    cardHeaderAccent: {
        color: theme.palette.white.main,
        overflow: 'hidden',
        whiteSpace: 'nowrap',
        textOverflow: 'ellipsis',
        backgroundColor: theme.palette.accent.main,
        borderRadius: '4px 4px 0px 0px',
        padding: '12px 24px',
    },
    fullHeight: {
        height: '100%',
    },
});

export class Cards extends Component {
    static propTypes = {
        title: PropTypes.any,
        primaryHeader: PropTypes.bool,
        accentHeader: PropTypes.bool,
        fullHeight: PropTypes.bool,
        noPadding: PropTypes.bool,
        noHeader: PropTypes.bool,
        children: PropTypes.any,
        classes: PropTypes.object.isRequired,
        help: PropTypes.object,
        customBackgroundColor: PropTypes.any,
        customTitleColor: PropTypes.any,
        customTitleBgColor: PropTypes.any,
        customTextColor: PropTypes.any,
        squareTop: PropTypes.bool,
        smallTitle: PropTypes.bool,
        standardCardId: PropTypes.string,
        subCard: PropTypes.bool,
        style: PropTypes.object,
    };

    render() {
        const {
            classes,
            title,
            help,
            children,
            primaryHeader,
            accentHeader,
            smallTitle = false,
            subCard = false,
            style = {},
        } = this.props;
        const customBG = !!this.props.customBackgroundColor
            ? { backgroundColor: this.props.customBackgroundColor }
            : null;
        const customTitleBG = !!this.props.customTitleBgColor
            ? { backgroundColor: this.props.customTitleBgColor }
            : null;
        const customTitle = !!this.props.customTitleColor ? { color: this.props.customTitleColor } : null;
        const customText = !!this.props.customTextColor ? { color: `${this.props.customTextColor} !important` } : null;
        const fullHeight = !!this.props.fullHeight ? { height: '100%' } : null;
        const squareTop = !!this.props.squareTop
            ? { borderTopLeftRadius: 0, borderTopRightRadius: 0 }
            : { borderTopLeftRadius: 4, borderTopRightRadius: 4 };
        const standardCardId = !!this.props.standardCardId
            ? this.props.standardCardId
            : `standard-card${typeof title === 'string' ? '-' + title.replace(/ /g, '-').toLowerCase() : ''}`;
        return (
            <Card
                data-testid={standardCardId}
                id={standardCardId}
                className={`${classes.card} StandardCard`}
                style={{ ...customBG, ...fullHeight, ...style }}
            >
                {!this.props.noHeader && (
                    <CardHeader
                        style={{ ...squareTop, ...customTitleBG, ...customTitle }}
                        title={title}
                        titleTypographyProps={{
                            variant: smallTitle ? 'h6' : 'h5',
                            component: subCard ? 'h4' : 'h3',
                            color: 'inherit',
                            'data-testid': `${standardCardId}-header`,
                        }}
                        action={!!help && !!help.text && <HelpIcon {...help} />}
                        classes={{
                            root:
                                (primaryHeader && classes.cardHeaderPrimary) ||
                                (accentHeader && classes.cardHeaderAccent),
                        }}
                    />
                )}
                <CardContent
                    data-testid={`${standardCardId}-content`}
                    className={`${(this.props.noPadding && classes.cardContentNoPadding) || ''}`}
                    style={{ ...customText }}
                >
                    {children}
                </CardContent>
            </Card>
        );
    }
}

const StyledCard = withStyles(styles, { withTheme: true })(Cards);
export const StandardCard = props => <StyledCard {...props} />;
export default StandardCard;
