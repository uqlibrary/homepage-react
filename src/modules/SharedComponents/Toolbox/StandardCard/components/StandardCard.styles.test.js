import { styles } from './StandardCard';

describe('AppLoader component', () => {
    it('should have a proper style generator', () => {
        const theme = {
            typography: {
                fontWeightRegular: 'test1',
            },
            palette: {
                white: {
                    main: 'test2',
                },
                primary: {
                    main: 'test3',
                },
                accent: {
                    main: 'test4',
                },
            },
        };
        expect(styles(theme)).toMatchSnapshot();
    });
});
