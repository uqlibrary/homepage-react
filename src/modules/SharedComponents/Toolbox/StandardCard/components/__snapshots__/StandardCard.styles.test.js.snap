// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLoader component should have a proper style generator 1`] = `
{
  "card": {
    "fontWeight": "test1",
    "overflow": "unset",
  },
  "cardContentNoPadding": {
    "paddingBottom": "0px !important",
    "paddingLeft": "0px !important",
    "paddingRight": "0px !important",
    "paddingTop": "0px !important",
  },
  "cardHeaderAccent": {
    "backgroundColor": "test3",
    "borderRadius": "4px 4px 0px 0px",
    "color": "test2",
    "overflow": "hidden",
    "padding": "12px 24px",
    "textOverflow": "ellipsis",
    "whiteSpace": "nowrap",
  },
  "cardHeaderPrimary": {
    "backgroundColor": "test4",
    "borderRadius": "4px 4px 0px 0px",
    "color": "test2",
    "overflow": "hidden",
    "padding": "12px 24px",
    "textOverflow": "ellipsis",
    "whiteSpace": "nowrap",
  },
  "fullHeight": {
    "height": "100%",
  },
}
`;
